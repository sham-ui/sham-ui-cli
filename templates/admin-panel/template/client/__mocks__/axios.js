const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            INTERCEPTORS.request( request );
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = {
    csrfToken: 'csrf-token',
    user: {
        Name: 'John Smith',
        Email: 'j.smith@example.com'
    }
};

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        this
            .use( 'get', '/csrftoken', '', 200, { 'x-csrf-token': defaultMocksData.csrfToken } )
            .use( 'get', '/validsession', defaultMocksData.user );
        return this;
    }
};
