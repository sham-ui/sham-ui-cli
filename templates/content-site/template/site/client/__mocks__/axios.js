const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};
const INTERCEPTORS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const request = {
                url,
                method,
                headers: {}
            };
            if ( INTERCEPTORS.request ) {
                INTERCEPTORS.request( request );
            }
            const { status, data, headers } = FIXTURES[ method ][ url ];
            if ( 200 === status ) {
                const response = {
                    data,
                    headers,
                    status
                };
                return Promise.resolve(
                    INTERCEPTORS.response.success( response )
                );
            } else {
                const error = {
                    config: {
                        url,
                        method,
                        baseURL: '/'
                    },
                    response: {
                        data,
                        headers,
                        status
                    }
                };
                return INTERCEPTORS.response.fail( error );
            }
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);


export default {
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        const instance  = {
            ...MOCKS,
            request( { url, method = 'get', data, params } ) {
                const args = [ url ];
                if ( undefined !== data ) {
                    args.push( data );
                }
                if ( undefined !== params ) {
                    args.push( params );
                }
                return instance[ method ].apply( null, args );
            },
            interceptors: {
                request: {
                    use( interceptor ) {
                        INTERCEPTORS.request = interceptor;
                    }
                },
                response: {
                    use( success, fail ) {
                        INTERCEPTORS.response = {
                            success,
                            fail
                        };
                    }
                }
            }
        };
        return instance;
    } ),
    use( method, url, data, status = 200, headers = {} ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"` );
        }
        FIXTURES[ method ][ url ] = {
            data,
            status,
            headers
        };
        return this;
    },

    useDefaultMocks() {
        return this;
    }
};
