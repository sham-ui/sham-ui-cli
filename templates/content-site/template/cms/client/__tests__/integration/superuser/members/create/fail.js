import setup, { app } from '../../../helpers';
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
} );

it( 'fail create member', async() => {
    expect.assertions( 1 );

    const DI = setup();
    axios
        .useDefaultMocks()
        .use( 'get', '/validsession', {
            ...axios.defaultMocksData.user,
            IsSuperuser: true
        } )
        .use( 'get', 'admin/members', {
            members: [],
            meta: {
                total: 0,
                offset: 0,
                limit: 50
            }
        } )
        .use( 'post', 'admin/members', {
            'Status': 'Bad request',
            'Messages': [ 'Can\'t create member' ]
        }, 500 );

    history.pushState( {}, '', 'http://client.example.com/members' );
    await app.start( DI );

    app.click( '[data-test-toggle-create-form]' );
    await app.waitRendering();

    app.form.fill( 'name', 'John Smith' );
    app.form.fill( 'email', 'john.smith@test.com' );
    app.form.fill( 'pass', 'test' );
    app.form.submit();

    app.click( '[data-test-modal] [data-test-ok-button]' );
    await app.waitRendering();
    app.checkMainPanel();
} );

it( 'fail create member (500 status code)', async() => {
    expect.assertions( 1 );

    const DI = setup();
    axios
        .useDefaultMocks()
        .use( 'get', '/validsession', {
            ...axios.defaultMocksData.user,
            IsSuperuser: true
        } )
        .use( 'get', 'admin/members', {
            members: [],
            meta: {
                total: 0,
                offset: 0,
                limit: 50
            }
        } )
        .use( 'post', 'admin/members', {}, 500 );

    history.pushState( {}, '', 'http://client.example.com/members' );
    await app.start( DI );

    app.click( '[data-test-toggle-create-form]' );
    await app.waitRendering();

    app.form.fill( 'name', 'John Smith' );
    app.form.fill( 'email', 'john.smith@test.com' );
    app.form.fill( 'pass', 'test' );
    app.form.submit();

    app.click( '[data-test-modal] [data-test-ok-button]' );
    await app.waitRendering();
    app.checkMainPanel();
} );
