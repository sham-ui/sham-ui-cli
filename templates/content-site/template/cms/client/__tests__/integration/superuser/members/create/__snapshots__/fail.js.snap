// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fail create member (500 status code) 1`] = `
"<h4 class=\\"page-header\\"> Members
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"members-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Hide create new member form</button>
      <div class=\\"panel mg-b-30 create-member\\">
        <form class=\\"form-layout\\">
          <!--0-->
          <ul class=\\"errors-panel\\">
            <li>Create member fail</li>
          </ul>
          <!--0-->
          <!--1-->
          <div class=\\"form-group\\">
            <label>Name:</label>
            <input name=\\"name\\" type=\\"text\\" autofocus=\\"\\">
          </div>
          <div class=\\"form-group\\">
            <label>Email:</label>
            <input name=\\"email\\" type=\\"text\\">
          </div>
          <div class=\\"form-group\\">
            <label></label>
            <label class=\\"checkbox\\">
              <input name=\\"is_superuser\\" type=\\"checkbox\\"><span>Is Superuser</span>
            </label>
          </div>
          <div class=\\"form-group\\">
            <label>Password:</label>
            <input name=\\"pass\\" type=\\"password\\">
          </div>
          <!--0-->
          <!--2-->
          <button type=\\"submit\\">
            <!--3-->Create
            <!--4-->
          </button>
          <!--5-->
        </form>
        <!--0-->
        <!--1-->
        <!--0-->
      </div>
      <!--1-->
    </div>
    <div class=\\"container\\">
      <div class=\\"members-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--0-->
            <table>
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Name</th>
                  <th>Email</th>
                  <th>Is superuser?</th>
                  <th></th>
                </tr>
              </thead>
              <tbody></tbody>
            </table>
            <!--0-->
            <!--0-->
            <!--0-->
            <!--1-->
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--2-->
          </div>
        </div>
      </div>
      <!--2-->
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->"
`;

exports[`fail create member 1`] = `
"<h4 class=\\"page-header\\"> Members
  <!--0-->
</h4>
<div class=\\"page-body\\">
  <div class=\\"members-page\\">
    <!--0-->
    <!--0-->
    <div class=\\"container\\">
      <button data-test-toggle-create-form=\\"\\" class=\\"mg-b-30\\">Hide create new member form</button>
      <div class=\\"panel mg-b-30 create-member\\">
        <form class=\\"form-layout\\">
          <!--0-->
          <ul class=\\"errors-panel\\">
            <li>Can't create member</li>
          </ul>
          <!--0-->
          <!--1-->
          <div class=\\"form-group\\">
            <label>Name:</label>
            <input name=\\"name\\" type=\\"text\\" autofocus=\\"\\">
          </div>
          <div class=\\"form-group\\">
            <label>Email:</label>
            <input name=\\"email\\" type=\\"text\\">
          </div>
          <div class=\\"form-group\\">
            <label></label>
            <label class=\\"checkbox\\">
              <input name=\\"is_superuser\\" type=\\"checkbox\\"><span>Is Superuser</span>
            </label>
          </div>
          <div class=\\"form-group\\">
            <label>Password:</label>
            <input name=\\"pass\\" type=\\"password\\">
          </div>
          <!--0-->
          <!--2-->
          <button type=\\"submit\\">
            <!--3-->Create
            <!--4-->
          </button>
          <!--5-->
        </form>
        <!--0-->
        <!--1-->
        <!--0-->
      </div>
      <!--1-->
    </div>
    <div class=\\"container\\">
      <div class=\\"members-list\\">
        <div class=\\"panel\\">
          <div class=\\"pagination\\">
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--0-->
            <table>
              <thead>
                <tr>
                  <th>ID</th>
                  <th>Name</th>
                  <th>Email</th>
                  <th>Is superuser?</th>
                  <th></th>
                </tr>
              </thead>
              <tbody></tbody>
            </table>
            <!--0-->
            <!--0-->
            <!--0-->
            <!--1-->
            <div class=\\"page-buttons\\">
              <!--0-->
              <button data-test-pagination-page=\\"1\\" disabled=\\"\\">1</button>
              <!--1-->
            </div>
            <!--2-->
          </div>
        </div>
      </div>
      <!--2-->
    </div>
  </div>
  <!--1-->
</div>
<!--0-->
<!--0-->
<!--0-->"
`;
